src/Toolkit/Xml.php

```diff
@@ -66,13 +64,13 @@ class Xml
 	/**
 	 * Generates a single attribute or a list of attributes
 	 *
-	 * @param string|array $name String: A single attribute with that name will be generated.
-	 *                           Key-value array: A list of attributes will be generated. Don't pass a second argument in that case.
+	 * @param string|array|null $name String: A single attribute with that name will be generated.
+	 *                                Key-value array: A list of attributes will be generated. Don't pass a second argument in that case.
 	 * @param mixed $value If used with a `$name` string, pass the value of the attribute here.
 	 *                     If used with a `$name` array, this can be set to `false` to disable attribute sorting.
 	 * @return string|null The generated XML attributes string
 	 */
-	public static function attr($name, $value = null): string|null
+	public static function attr(string|array|null $name, $value = null): string|null
```

Comment:
How can `$name` be `null`? Won't this break on string concatenation? Also, what would the use case be to pass `null`?

Reply:
I remember again - our unit tests start failing a lot when not allowing for `null`: https://github.com/getkirby/kirby/actions/runs/3786702322/jobs/6437840246

Reply:
I assume those currently work, because they also get caught by `$value === null`

Reply:
All of these erroneous calls come from `Toolkit/Xml.php:373` (`Xml::tag()`, which accepts `null` for the `$attr` argument).

I'm pretty sure there is no reason to support `null` for this argument either. So I think we should change the default to this:

```diff
- public static function tag(string $name, $content = '', array $attr = null
+ public static function tag(string $name, $content = '', array $attr = []
```

Reply:
There needed to be one additional line changed which calls `::tag()` with all arguments. But I think then it works. Can you have a look? And then we should add it as breaking change.